core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -305,25 +316,46 @@ private boolean notifySubscriber(Subscriber<Object> subscriber, Queue<Object> qu
 
             if (demand == Long.MAX_VALUE || demandUpdater.compareAndSet(this, demand, demand - 1)) {
                 @SuppressWarnings("unchecked")
-                final T o = (T) queue.remove();
+                T o = (T) queue.remove();
+                ReferenceCountUtil.touch(o);
                 onRemoval(o);
-                if (!subscription.withPooledObjects() && o instanceof ByteBufHttpData) {
-                    ByteBuf buf = ((ByteBufHttpData) o).buf();
-                    try {
-                        subscriber.onNext(HttpData.of(ByteBufUtil.getBytes(buf)));
-                    } finally {
-                        buf.release();
+                if (!subscription.withPooledObjects()) {
+                    if (o instanceof ByteBufHolder) {
```

Comment:
Oh, there's a safeRelease added up above at !isOpen which is totally new, that's the fix - I got it now, sorry for the noise :)